creator-node/src/snapbackSM/peerSetManager.js

```diff
@@ -250,6 +283,104 @@ class PeerSetManager {
+   */
+  async updateEndpointToSpIdMap (ethContracts) {
+    let endpointToSPIdMap = {}
+    try {
+      const contentNodes = await ethContracts.getServiceProviderList('content-node')
+      contentNodes.forEach(cn => { endpointToSPIdMap[cn.endpoint] = cn.spID })
+    } catch (e) {
+      this.logError(`[updateEndpointToSpIdMap]: ${e.message}`)
+    }
+
+    if (Object.keys(endpointToSPIdMap).length > 0) this.endpointToSPIdMap = endpointToSPIdMap
+    if (Object.keys(this.endpointToSPIdMap).length === 0) {
+      const errorMsg = '[updateEndpointToSpIdMap]: Unable to initialize this.endpointToSPIdMap'
+      this.logError(errorMsg)
+      throw new Error(errorMsg)
+    }
+  }
+
+  /**
+   * Converts provided array of nodeUser info to issue to a map(replica set node => userWallets[]) for easier access
+   *
+   * @param {Array} nodeUsers array of objects with schema { user_id, wallet, primary, secondary1, secondary2 }
+   * @returns {Object} map of replica set endpoint strings to array of wallet strings of users with that node as part of replica set
+   */
+  buildReplicaSetNodesToUserWalletsMap (nodeUsers) {
+    const replicaSetNodesToUserWalletsMap = {}
+
+    nodeUsers.forEach(userInfo => {
+      const { wallet, primary, secondary1, secondary2 } = userInfo
+      const replicaSet = [primary, secondary1, secondary2]
+
+      replicaSet.forEach(node => {
+        if (!replicaSetNodesToUserWalletsMap[node]) {
+          replicaSetNodesToUserWalletsMap[node] = []
+        }
+
+        replicaSetNodesToUserWalletsMap[node].push(wallet)
+      })
+    })
+
+    return replicaSetNodesToUserWalletsMap
+  }
+
+  // ============== `this.unhealthyPrimaryToWalletMap` functions ==============
+
+  /**
+   * Perform a simple health check to see if a primary is truly unhealthy. If the primary returns a
+   * non-200 response, track the timestamp in the map. If the health check has failed for a primary over
+   * `this.maxNumberSecondsPrimaryRemainsUnhealthy`, return as unhealthy. Else, keep track of the timestamp
+   * of the visit if not already tracked.
+   *
+   * If the primary is healthy, reset the counter in the map and return as healthy.
+   * @param {string} primary primary endpoint
+   * @returns boolean of whether primary is healthy or not
+   */
+  async isPrimaryHealthy (primary) {
+    const isHealthy = await this.isNodeHealthy(primary, true)
+
+    if (!isHealthy) {
+      const failedTimestamp = this.getEarliestFailedHealthCheckTimestamp(primary)
+
+      if (failedTimestamp) {
+        // Generate the date of the failed timestamp + max hours threshold
+        let failedTimestampPlusThreshold = new Date(failedTimestamp)
+        failedTimestampPlusThreshold.setSeconds(
+          failedTimestamp.getSeconds() + this.maxNumberSecondsPrimaryRemainsUnhealthy
+        )
+
+        // Determine if the failed timestamp + max hours threshold surpasses our allowed time threshold
+        const now = new Date()
```

Comment:
I dont think timezones will be a problem. the Date object should be able to internally compare timestamps correctly. https://stackoverflow.com/a/27271132

Reply:
yeah timezone not problem or our entire system would die lol

Reply:
lol I literally laughed out loud

Reply:
fwiw this comment wasn't referring to intra-node date comparisons but more in the vein of metrics / timestamps for network wide comparisons

Reply:
ahhh i see your point

Reply:
since nodes in diff time zones will have their date's "day" change at diff times. so we'd have situation where some nodes are 10hrs into their day and they see another node with very little data in beginning of their day

instinctively i'm not sure if there's any issue with this? can you think of any problematic scenarios?

Reply:
the only problem i can see is if we expose data around reconfig submissions and end up having to calculate and convert timezones to compare them manually vs. having everything in UTF. we can address post metrics if need be

Reply:
hmmm yeah this is probably not great - we do this in a couple other places too, but prob not all. inconsistency prob won't help us. i dn't think this is a crazy problem though, but i'll make a note to handle separately soon after. cool with you?

Reply:
yeah, let's not block on this  just wanted to point it out